font/src/ft/fc.rs

```diff
@@ -11,42 +11,34 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-//
-use std::fmt;
-use std::ptr;
-
+use std::{fmt, ptr};
+
+use fontconfig::fontconfig::{
+    FcResultNoMatch, {FcFontList, FcFontMatch, FcFontSort},
+    {FcMatchFont, FcMatchPattern, FcMatchScan}, {FcSetApplication, FcSetSystem},
+    {FC_SLANT_ITALIC, FC_SLANT_OBLIQUE, FC_SLANT_ROMAN},
+    {FC_WEIGHT_BLACK, FC_WEIGHT_BOLD, FC_WEIGHT_EXTRABLACK, FC_WEIGHT_EXTRABOLD},
+    {FC_WEIGHT_BOOK, FC_WEIGHT_MEDIUM, FC_WEIGHT_REGULAR, FC_WEIGHT_SEMIBOLD},
+    {FC_WEIGHT_EXTRALIGHT, FC_WEIGHT_LIGHT, FC_WEIGHT_THIN},
+};
 use foreign_types::{ForeignType, ForeignTypeRef};
 
-use fontconfig::fontconfig as ffi;
-
-use self::ffi::FcResultNoMatch;
-use self::ffi::{FcFontList, FcFontMatch, FcFontSort};
-use self::ffi::{FcMatchFont, FcMatchPattern, FcMatchScan};
-use self::ffi::{FcSetApplication, FcSetSystem};
-use self::ffi::{FC_SLANT_ITALIC, FC_SLANT_OBLIQUE, FC_SLANT_ROMAN};
-use self::ffi::{FC_WEIGHT_BLACK, FC_WEIGHT_BOLD, FC_WEIGHT_EXTRABLACK, FC_WEIGHT_EXTRABOLD};
-use self::ffi::{FC_WEIGHT_BOOK, FC_WEIGHT_MEDIUM, FC_WEIGHT_REGULAR, FC_WEIGHT_SEMIBOLD};
-use self::ffi::{FC_WEIGHT_EXTRALIGHT, FC_WEIGHT_LIGHT, FC_WEIGHT_THIN};
-
-pub mod config;
-pub use self::config::{Config, ConfigRef};
+pub use char_set::{CharSet, CharSetRef};
+pub use config::{Config, ConfigRef};
+pub use font_set::{FontSet, FontSetRef};
+pub use object_set::{ObjectSet, ObjectSetRef};
+pub use pattern::{Pattern, PatternRef};
```

Comment:
This is done incorrectly in the latest master too, but generally we do not want to use `pub use` excessively. Imports like `self::Pattern` make it extremely difficult to understand where things are coming from. Meanwhile `pattern::Pattern` is without a doubt easier to understand.

Reply:
I've got no problem with that, just trying to preserve the existing paths. Do you want me to refactor the visibility in this merge request? Because I was already getting the feeling that I had touched too many files.

Reply:
Feel free to do either, I'd understand not wanting to touch any existing code if the goal is just to transition to a new edition.

However when it comes to style improvements, I'm always willing to make exceptions since these should just improve codestyle in general.